tiledb/sm/array_schema/domain.h

```diff
@@ -74,20 +76,15 @@ class Domain {
   /*     CONSTRUCTORS & DESTRUCTORS    */
   /* ********************************* */
 
-  /** Empty constructor. */
-  Domain();
```

Comment:
Don't forget to delete the default constructor as it'll make the error messages more clear if someone accidentally adds code that attempts to use it.